test/e2e/azure_test.go

```diff
@@ -33,7 +33,7 @@ import (
 	"sigs.k8s.io/cluster-api/util"
 )
 
-var _ = Describe("Workoad cluster creation", func() {
+var _ = Describe("Workload cluster creation", func() {
```

Comment:
good catch!

needs an update to https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml#L84 as well 

I recommend doing the spelling fix in a separate PR so it can merge quickly and not impact other PRs when we merge the test-infra config fix